bftengine/src/bftengine/Digest.cpp

```diff
@@ -13,11 +13,87 @@
 
 #include <string.h>
 #include <stdio.h>
-#include "Crypto.hpp"
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
+#include <cryptopp/dll.h>
+#include <cryptopp/pem.h>
+#pragma GCC diagnostic pop
+
+#include "DigestType.h"
+#include "assertUtils.hpp"
+#include <cryptopp/cryptlib.h>
+#include <cryptopp/ida.h>
+#include <cryptopp/eccrypto.h>
+
+using namespace CryptoPP;
+using namespace std;
+
+#if defined MD5_DIGEST
+#include <cryptopp/md5.h>
+#define DigestType Weak1::MD5
+#elif defined SHA256_DIGEST
+#define DigestType SHA256
+#elif defined SHA512_DIGEST
+#define DigestType SHA512
+#endif
 
 namespace bftEngine {
 namespace impl {
 
+size_t DigestUtil::digestLength() { return DigestType::DIGESTSIZE; }
```

Comment:
Should it be a part of crypto_utils?

Reply:
It is used only by bftengine, so I would prefer not to touch it now